pyinterprod/uniprot/unirule.py

```diff
@@ -707,3 +707,214 @@ def update_signatures(filepath: str, uri: str):
     con.commit()
     cur.close()
     con.close()
+
+
+MAX_DOM_BY_GROUP = 20
+DOM_OVERLAP_THRESHOLD = 0.3
+REPR_DOM_DATABASES = ["H", "J", "M", "R", "N"]
+
+
+def get_repr_domains(ora_url: str, output: str = "repr_domains.tsv"):
+    con = oracledb.connect(ora_url)
+    cur = con.cursor()
+
+    dbcodes = ', '.join([f"'{db}'" for db in REPR_DOM_DATABASES])
+    cur.execute(
+        f"""
+        SELECT PROTEIN_AC, H.METHOD_AC, H.DBCODE, POS_FROM, POS_TO, FRAGMENTS
+        FROM INTERPRO.MATCH H
+        INNER JOIN INTERPRO.METHOD D
+        ON H.METHOD_AC = D.METHOD_AC
+        WHERE H.DBCODE in ({dbcodes})
+        AND (D.SIG_TYPE = 'D' OR D.SIG_TYPE = 'R')
+        ORDER BY PROTEIN_AC
+        """
+    )
```

Comment:
You're right my last suggestion had a mistake in it, as I forgot that Oracle does not support binding a Python tuple to an SQL `IN` clause directly :sweat_smile: 

It might be better to avoid a direct injection of the values in `REPR_DOM_DATABASES` into the query. You can do this with dynamically creating a string of placeholders for the `IN` clause. E.g:
```suggestion
    placeholders = ', '.join(':' + str(i + 1) for i in range(len(REPR_DOM_DATABASES)))
    params = {str(i + 1): db for i, db in enumerate(REPR_DOM_DATABASES)}
    cur.execute(
        f"""
        SELECT PROTEIN_AC, H.METHOD_AC, H.DBCODE, POS_FROM, POS_TO, FRAGMENTS
        FROM INTERPRO.MATCH H
        INNER JOIN INTERPRO.METHOD D
        ON H.METHOD_AC = D.METHOD_AC
        WHERE H.DBCODE in ({placeholders})
        AND (D.SIG_TYPE = 'D' OR D.SIG_TYPE = 'R')
        ORDER BY PROTEIN_AC
        """, params
    )
   
```

Reply:
Ah! Even better, thank you!